src/app/[locale]/viewer/_components/SubmodelsOverviewCard.tsx

```diff
@@ -26,115 +26,123 @@ export function SubmodelsOverviewCard(props: SubmodelsOverviewCardProps) {
-                    endpoint: submodelDescriptor.endpoints[0].protocolInformation.href,
-                });
-            });
+        let submodelsPromise;
+
+        if (registryAasData && registryAasData.submodelDescriptors) {
+            // Fetch submodel from provided endpoint
+             submodelsPromise = Promise.all(
+                registryAasData.submodelDescriptors.map(async (submodelDescriptor) => {
+                    const endpoint = submodelDescriptor?.endpoints[0].protocolInformation.href;
+
+                    if (endpoint) {
+                        const submodelData = await getSubmodelFromSubmodelDescriptor(endpoint);
+                        return {
+                            id: submodelDescriptor.id,
+                            label: submodelDescriptor.idShort ?? '',
+                            submodelData: submodelData,
+                        };
+                    }
+                    return null;
+                }).filter(item => !!item)
+            );
         } else {
-            for (const reference of props.smReferences as Reference[]) {
-                try {
-                    const submodelFromRegistry = env.SUBMODEL_REGISTRY_API_URL
-                        ? await submodelRegistryServiceClient.getSubmodelDescriptorsById(reference.keys[0].value)
-                        : null;
-                    submodels.push({
-                        id: submodelFromRegistry.id,
-                        label: submodelFromRegistry.idShort ?? '',
-                        endpoint: submodelFromRegistry.endpoints[0].protocolInformation.href,
-                    });
-                } catch (e) {
-                    // Submodel registry is not available or submodel not found there -> search in repo
-                    if (e instanceof TypeError || (e instanceof Response && e.status === 404)) {
-                        await fetchSubmodelFromRepo(reference);
-                    } else {
-                        console.error(e);
+            // Search in default registry
+            submodelsPromise = Promise.all(
+                (props.smReferences as Reference[]).map(async (reference) => {
+                    let submodelData;
+                    try {
+                        const submodelDescriptor = env.SUBMODEL_REGISTRY_API_URL
+                            ? await submodelRegistryServiceClient.getSubmodelDescriptorsById(reference.keys[0].value)
+                            : null;
+                        const endpoint = submodelDescriptor?.endpoints[0].protocolInformation.href;
+
+                        if (endpoint) {
+                            submodelData = await getSubmodelFromSubmodelDescriptor(endpoint);
+                            submodelData = {
+                                id: submodelDescriptor.id,
+                                label: submodelDescriptor.idShort ?? '',
+                                submodelData: submodelData
+                            };
```

Comment:
why is the `submodelData` overridden instantly, looks strange
maybe rename one variable, as it is not the same type

Reply:
maybe move the mapping method in a separate method even, so we can just `return {id: ..., label:..., submodelData:...}`

would work with the other `submodelData` assignments as well